src/editor/reducers/notebook-reducer.js

```diff
@@ -79,13 +79,26 @@ const notebookReducer = (state = newNotebook(), action) => {
     }
 
     case "GOT_NOTEBOOK_REVISION_LIST": {
-      const { revisionList, selectedRevisionId } = action;
+      const { revisionContent, revisionList, selectedRevisionId } = action;
+
+      const previousRevisionContent =
+        revisionList.length > 0 ? revisionContent[revisionList[0].id] : "";
+
+      let passedRevisionId = selectedRevisionId;
+
+      if (
+        selectedRevisionId === undefined &&
+        state.iomd === previousRevisionContent
+      ) {
+        passedRevisionId = revisionList[0].id;
+      }
+
       return Object.assign({}, state, {
         notebookHistory: {
           ...(state.notebookHistory || {}),
           revisionList,
           revisionListFetchStatus: "IDLE",
-          selectedRevisionId
+          passedRevisionId
```

Comment:
I think you may have forgotten to commit some other changes? I don't see passedRevisionId (or any changed logic to use it) in the other files.

Also, I think the right thing here is to calculate `haveLocalChanges` in this reducer. I'm actually not sure what passedRevisionId is supposed to represent.